package/kernel/mwlwifi/patches/006-remove-uaccess-and-get_fs-calls-from-PCIe-for-Kenel-.patch

```diff
@@ -0,0 +1,49 @@
+From 30db27ba3df53cb763de0411c805c49df9448742 Mon Sep 17 00:00:00 2001
+From: Stefan Kalscheuer <stefan@stklcode.de>
+Date: Sun, 18 Jun 2023 17:53:27 +0200
+Subject: [PATCH 1/2] remove uaccess and get_fs calls from PCIe for Kenel >=
+ 5.18
+
+Remove the calls to deprecated get_fs and force_uaccess_* API for modern
+kernels.
+
+Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
+---
+ hif/pcie/pcie.c | 6 ++++--
+ 1 file changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/hif/pcie/pcie.c b/hif/pcie/pcie.c
+index 24453b6..bee1cc5 100644
+--- a/hif/pcie/pcie.c
++++ b/hif/pcie/pcie.c
+@@ -1294,7 +1294,9 @@ static void pcie_bf_mimo_ctrl_decode(struct mwl_priv *priv,
+ 	const char filename[] = "/tmp/BF_MIMO_Ctrl_Field_Output.txt";
+ 	char str_buf[256];
+ 	char *buf = &str_buf[0];
++#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
+ 	mm_segment_t oldfs;
++#endif
+ 
+ #if LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0)
+ 	oldfs = get_fs();
+@@ -1302,7 +1304,7 @@ static void pcie_bf_mimo_ctrl_decode(struct mwl_priv *priv,
+ #elif LINUX_VERSION_CODE < KERNEL_VERSION(5,10,0)
+ 	oldfs = get_fs();
+ 	set_fs(KERNEL_DS);
+-#else
++#elif LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
+ 	oldfs = force_uaccess_begin();
+ #endif
+ 
+@@ -1326,7 +1328,7 @@ static void pcie_bf_mimo_ctrl_decode(struct mwl_priv *priv,
+ 
+ #if LINUX_VERSION_CODE < KERNEL_VERSION(5,10,0)
+ 	set_fs(oldfs);
+-#else
++#elif LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
+ 	force_uaccess_end(oldfs);
```

Comment:
Is calling `force_uaccess_begin` and `force_uaccess_end` not needed at all any more?

Reply:
`force_uaccess_{begin,end}` were introduced for transitional purpose to get rid of `get_fs()` calls in 5.10 (https://github.com/torvalds/linux/commit/3d13f313ce4c34c524ccc37986fe77172f601ff3) followed by the introduction of `CONFIG_SET_FS` to migrate one arch after another (https://github.com/torvalds/linux/commit/5e6e9852d6f76e01b2e6803c74258afa5b432bc5)

The internal read/write logic has been refactored, s.t. driver code can transparently operate on userspace/kernel pointers without messing with boundaries individually (mainly https://github.com/torvalds/linux/commit/4d03e3cc59828c82ee89ea6e27a2f3cdf95aaadf and https://github.com/torvalds/linux/commit/36e2c7421f02a22f71c9283e55fdb672a9eb58e7).

Finally, `CONFIG_SET_FS` was removed in 5.18 for all architectures ([ChangeLog-5.18](https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.18)) and the entire `{g,s}et_fs()` and `force_uaccess_{begin,end}` logic removed in https://github.com/torvalds/linux/commit/967747bbc084b93b54e66f9047d342232314cd25

(some additional reading on the evolution of `set_fs()` can be found here: https://lwn.net/Articles/832121/)

----

So, if I didn't miss anything, the calls should not require any replacement.
I'll try to add some compact and hopefully meaningful wording to the patch description.